qa/logger-usage/src/main/java/org/elasticsearch/qa/loggerusage/ESLoggerUsageChecker.java

```diff
@@ -0,0 +1,437 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.qa.loggerusage;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.IntInsnNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.LineNumberNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.tree.TypeInsnNode;
+import org.objectweb.asm.tree.analysis.Analyzer;
+import org.objectweb.asm.tree.analysis.AnalyzerException;
+import org.objectweb.asm.tree.analysis.BasicInterpreter;
+import org.objectweb.asm.tree.analysis.BasicValue;
+import org.objectweb.asm.tree.analysis.Frame;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class ESLoggerUsageChecker {
+    public static final String LOGGER_CLASS = "org.elasticsearch.common.logging.ESLogger";
+    public static final String THROWABLE_CLASS = "java.lang.Throwable";
+    public static final List<String> LOGGER_METHODS = Arrays.asList("trace", "debug", "info", "warn", "error");
+
+    public static void main(String... args) throws Exception {
+        System.out.println("checking for wrong usages of ESLogger...");
+        boolean[] wrongUsageFound = new boolean[1];
+        checkLoggerUsage(wrongLoggerUsage -> {
+            System.err.println(wrongLoggerUsage.getErrorLines());
+            wrongUsageFound[0] = true;
+        }, args);
+        if (wrongUsageFound[0]) {
+            throw new Exception("Wrong logger usages found");
+        } else {
+            System.out.println("No wrong usages found");
+        }
+    }
+
+    private static void checkLoggerUsage(Consumer<WrongLoggerUsage> wrongUsageCallback, String... classDirectories)
+        throws IOException {
+        for (String classDirectory : classDirectories) {
+            File dir = new File(classDirectory);
+            if (dir.isDirectory()) {
```

Comment:
Something like:
```groovy
if (false == (dir.exists() && dir.isDirectory())) {
  throw new InvalidUserDataException(dir + ' should be an existing directory.');
}
```

As in, should we just fail if the user points it as some garbage directory?